src/plugins/global-settings/colors/index.js

```diff
@@ -0,0 +1,360 @@
+							newAttributes.textColor = undefined
+							const appliedColor = find( colorsBeforeReset, color => color.slug === `stk-global-color-${ colorVarID }` )
+							newAttributes.style.color.text = appliedColor ? appliedColor.fallback || '#000000' : '#000000'
+						}
+					}
+				}
+
+				// Update the block attributes.
+				updateBlockAttributes( clientId, newAttributes )
+			} else {
+				const updatedAttributes = replaceGlobalColorAttributes( block.attributes, colorsBeforeReset, colorsAfterReset )
+
+				// This is used for pullquote compatibility.
+				if ( updatedAttributes.textColor ) {
+					const defaultColor = find( colorsAfterReset, updatedColor => updatedColor.slug === updatedAttributes.textColor )
+					if ( ! defaultColor ) {
+						const appliedColor = find( colorsBeforeReset, color => color.slug === updatedAttributes.textColor )
+						updatedAttributes.customTextColor = appliedColor ? appliedColor.fallback || '#000000' : '#000000'
+						updatedAttributes.textColor = undefined
+					} else {
+						updatedAttributes.customTextColor = defaultColor.color || '#000000'
+						updatedAttributes.textColor = undefined
+					}
+				}
+
+				// Update the block attributes.
+				if ( ! isEqual( updatedAttributes, block.attributes ) ) {
+					updateBlockAttributes( clientId, updatedAttributes )
+				}
+			}
+		} else if ( name.includes( 'kadence/' ) ) {
+			//
+			/**
+			 * For kadence blocks.
+			 * Similar to core blocks. If a kadence block uses a
+			 * color palette included in the theme or one of their
+			 * global palette, it uses the slug name as its attribute
+			 * (e.g. color: "kb-palette-1") and also adding a colorClass
+			 * attribute which will be applied as a custom css class
+			 * for that block. So in some cases, we have to set the colorClass
+			 * attribute to an empty string to avoid adding it to front end.
+			 */
+			const updatedAttributes = replaceGlobalColorAttributes( block.attributes, colorsBeforeReset, colorsAfterReset )
+
+			// Some kadence blocks e.g. Advanced Heading has colorClass attribute which will be appended to the h1 tag.
+			// Remove this to apply the hex color.
+			if ( updatedAttributes.colorClass ) {
+				updatedAttributes.colorClass = ''
+			}
+
+			// Update the block attributes.
+			if ( ! isEqual( updatedAttributes, block.attributes ) ) {
+				updateBlockAttributes( clientId, updatedAttributes )
+			}
+		} else {
+			// Default attributes adjustment. We can add more compatibility adjustments here.
+			const updatedAttributes = replaceGlobalColorAttributes( block.attributes, colorsBeforeReset, colorsAfterReset )
+
+			// Update the block attributes.
+			if ( ! isEqual( updatedAttributes, block.attributes ) ) {
+				updateBlockAttributes( clientId, updatedAttributes )
+			}
+		}
+	} )
+} )
+
+domReady( () => {
+	// Mount a style in the document
+	const globalStyleEl = document.createElement( 'style' )
+	globalStyleEl.setAttribute( 'id', 'stackable-global-colors' )
```

Comment:
Instead of creating an element and querying it via id, make this into a Component instead. Please check the `editor-loader.js` in global typography for an example.